When side-by-side builds of Visual Studio are installed, build is picking the wrong one

Unsure if this a build.exe issue or a Razzle issue. Based on experiments, Razzle appears to be okay.

# Environment
Visual Studio 16.2.0 Preview 4.0 [29018.107.d16.2stg] (installed to C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\)
Visual Studio 16.1 (installed to C:\Program Files (x86)\Microsoft Visual Studio\2019\RTM\)

# Steps to reproduce

1. From a _Developer Command Prompt For VS 2019 Preview_, run: 

```
> .\tools\razzle.cmd
> bcz
```

# Expected behavior

Build to succeed.

# Actual behavior

Build fails with:

```
Starting build...
MSBuild auto-detection: using msbuild version '16.200.19.31001' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\16.2\MSBuild\Current\bin'.
All packages listed in packages.config are already installed.
Skipping building appx...
"C:\Program Files (x86)\Microsoft Visual Studio\2019\16.1\MSBuild\Current\Bin\MSBuild.exe" E:\terminal\OpenConsole.sln /t:Clean,Build /m /p:Configuration=Debug /p:Platform=x64 /p:AppxBundle=false
Microsoft (R) Build Engine version 16.1.76+g14b0a930a7 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 6/19/2019 10:58:30 AM.
     1>Project "E:\terminal\OpenConsole.sln" on node 1 (Clean;Build target(s)).
     1>ValidateSolutionConfiguration:
         Building solution configuration "Debug|x64".
     1>Project "E:\terminal\OpenConsole.sln" (1) is building "E:\terminal\src\cascadia\TerminalApp\TerminalApp.vcxproj.
       metaproj" (2) on node 1 (Clean target(s)).
     2>Project "E:\terminal\src\cascadia\TerminalApp\TerminalApp.vcxproj.metaproj" (2) is building "E:\terminal\src\cas
       cadia\TerminalSettings\TerminalSettings.vcxproj" (6) on node 2 (Clean target(s)).
     6>E:\terminal\src\cppwinrt.build.pre.props(45,3): error MSB4019: The imported project "C:\Program Files (x86)\Micr
       osoft Visual Studio\2019\16.1\MSBuild\Microsoft\VC\v160\Microsoft.Cpp.Default.props" was not found. Confirm that
       the path in the <Import> declaration is correct, and that the file exists on disk. [E:\terminal\src\cascadia\Ter
       minalSettings\TerminalSettings.vcxproj]
```

The build is failing because its using/looking in the wrong location for these targets files. 

The first line says that its picked up the correct MSBuild ("16.2"):

```
MSBuild auto-detection: using msbuild version '16.200.19.31001' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\16.2\MSBuild\Current\bin'.
```

But a few lines down, you can see that its picking up MSBuild from the 16.1 folder:
```
"C:\Program Files (x86)\Microsoft Visual Studio\2019\16.1\MSBuild\Current\Bin\MSBuild.exe" E:\terminal\OpenConsole.sln /t:Clean,Build /m /p:Configuration=Debug /p:Platform=x64 /p:AppxBundle=false
```

--- a/tools/razzle.cmd
+++ b/tools/razzle.cmd
@@ -23,6 +23,16 @@ set PATH=%PATH%%OPENCON%\dep\nuget;
 rem Run nuget restore so you can use vswhere
 nuget restore %OPENCON%\OpenConsole.sln -Verbosity quiet
 
+:FIND_MSBUILD
+set MSBUILD=
+
+rem GH#1313: If msbuild is already on the path, we don't need to look for it.
+for %%X in (msbuild.exe) do (set MSBUILD=%%~$PATH:X)
+if defined MSBUILD (
+    echo Using MsBuild at %MSBUILD% which was already on the path.
+    goto :FOUND_MSBUILD
+)
+
 rem Find vswhere
 rem from https://github.com/microsoft/vs-setup-samples/blob/master/tools/vswhere.cmd
 for /f "usebackq delims=" %%I in (`dir /b /aD /o-N /s "%~dp0..\packages\vswhere*" 2^>nul`) do (
@@ -37,21 +47,22 @@ if not defined VSWHERE (
 )
 
 rem Add path to MSBuild Binaries
-for /f "usebackq tokens=*" %%B in (`%VSWHERE% -latest -products * -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe 2^>nul`) do (
+rem
+rem We're going to always prefer prerelease version of VS. This lets people who
+rem are using VS 2022 use that from the commandline over the 2019 version. This
+rem will use whatever the newest version of VS is, regardless if it's stable or
+rem not.
+rem
+for /f "usebackq tokens=*" %%B in (`%VSWHERE% -latest -prerelease -products * -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe 2^>nul`) do (
     set MSBUILD=%%B
-)
-
-rem Try to find MSBuild in prerelease versions of MSVS
-if not defined MSBUILD (
-    for /f "usebackq tokens=*" %%B in (`%VSWHERE% -latest -prerelease -products * -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe 2^>nul`) do (
-        set MSBUILD=%%B
-    )
 )
 
 if not defined MSBUILD (
     echo Could not find MsBuild on your machine. Please set the MSBUILD variable to the location of MSBuild.exe and run razzle again.
     goto :EXIT
 )
+
+:FOUND_MSBUILD
 
 set PATH=%PATH%%MSBUILD%\..;
 

